src/Perpetuum/Services/Channels/ChatCommands/AdminCommandHandlers.cs

```diff
@@ -623,39 +623,51 @@ public static void ZoneAddDecor(AdminCommandData data)
             if (!IsDevModeEnabled(data))
                 return;
 
-            bool err = false;
-            err = !int.TryParse(data.Command.Args[0], out int definition);
-            err = !int.TryParse(data.Command.Args[1], out int x);
-            err = !int.TryParse(data.Command.Args[2], out int y);
-            err = !int.TryParse(data.Command.Args[3], out int z);
-            err = !double.TryParse(data.Command.Args[4], out double qx);
-            err = !double.TryParse(data.Command.Args[5], out double qy);
-            err = !double.TryParse(data.Command.Args[6], out double qz);
-            err = !double.TryParse(data.Command.Args[7], out double qw);
-            err = !double.TryParse(data.Command.Args[8], out double scale);
-            err = !int.TryParse(data.Command.Args[9], out int cat);
-
-            if (err)
+            int? zoneId = data.Sender.ZoneId;
+            int definition;
+            int x;
+            int y;
+            int z;
+            double qx;
+            double qy;
+            double qz;
+            double qw;
+            double scale;
+            int cat;
+
+            try
+            {
+                //Validate Parameters
+                if(zoneId == null)
+                {
+                    throw new ArgumentNullException("zoneId is required for ZoneAddDecor command"); 
```

Comment:
`data.Sender.ZoneId` is a nullable int, but in this context I think it's required to be set.
Seeing how the command string used to look like `"zoneDecorAdd:zone_{0}:{1}", data.Sender.ZoneId,` I'm assuming it needs a value set

Reply:
yes